core/multicluster/multicluster.go

```diff
@@ -0,0 +1,92 @@
+package multicluster
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+	"github.com/weaveworks/weave-gitops/pkg/server/auth"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -generate
```

Comment:
🙌  yay love counterfeiter! so much better that the disaster that is gomock/mockgen